source/grid/grid_tools.cc

```diff
@@ -864,6 +864,17 @@ namespace GridTools
     const std::vector<Point<spacedim>> &all_vertices,
     std::vector<CellData<dim>> &        cells)
   {
+    // This only works for quads and hexes,
+    // and triangules. From the examples we tested,
+    // tetrahedron meshes loaded from Gmsh don't have
+    // this problem, so we skip them for now.
+
+    // Since we use std::abs in computing the measures
+    // of tetrahdrons, the measures are always positive.
+    // But if the orientation is wrong, we may get negative
+    // determinants of the Jacobians in MappingFE, which
+    // will triger the assert there.
```

Comment:
It's not clear to me what this comment refers to, in particular the "This only works..." at the very beginning. Are you talking about the algorithms used here, the whole function, or something else?

You're also mentioning a call to `abs()`, but there is none in the following several lines of code.

Reply:
About "This...": One of the things my wife always complains about when she looks at my writing is the use of "this" on its own. One typically uses it to reference something in the previous sentence, but it's not always clear what that is. She always requests that I write "This algorithm...", "this observation...", or whatever else it is that I refer to.

Here, you have no previous sentence at all, so it is doubly unclear what is meant :-)